src/progaudi.tarantool/Converters/ExecuteSqlRequest.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using ProGaudi.MsgPack.Light;
+using ProGaudi.Tarantool.Client.Model.Enums;
+using ProGaudi.Tarantool.Client.Model.Requests;
+
+namespace ProGaudi.Tarantool.Client.Converters
+{
+    internal class ExecuteSqlRequestConverter : IMsgPackConverter<ExecuteSqlRequest>
+    {
+        private MsgPackContext _context;
+        private bool _initialized;
+        private IMsgPackConverter<object> _nullConverter;
+        private IMsgPackConverter<string> _stringConverter;
+        private IMsgPackConverter<Key> _keyConverter;
+
+        public void Initialize(MsgPackContext context)
+        {
+            _context = context;
+        }
+
+        public void Write(ExecuteSqlRequest value, IMsgPackWriter writer)
+        {
+            if (!_initialized)
+            {
+                InitializeIfNeeded();
```

Comment:
Move `_initialized` check inside `InitializeIfNeeded()` method or rename it to `Initialize()`.

Reply:
It'll break any possibility of inlining that method, So no. And this is a stub of method which should be autogenerated by our generators in MsgPack library.

Reply:
I dont mind about leaving condition checking outside of the method. But then your method should be called `Initialize` (instead of `InitializeIfNeeded`). Because it is always perform initialization.